src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala

```diff
@@ -183,6 +185,15 @@ object TaskGroupBuilder extends StrictLogging {
       computeHealthCheck(healthCheck, portAssignments).foreach(builder.setHealthCheck)
     }
 
+    for {
+      lc <- container.lifecycle
+      killGracePeriodSeconds <- lc.killGracePeriodSeconds
+    } {
+      val durationInfo = DurationInfo.newBuilder.setNanoseconds((killGracePeriodSeconds * 1000000000).toLong)
```

Comment:
Nit: I like to have Java do the transformation `Duration.ofSeconds(killGracePeriodSeconds).getNano()`. This avoids magic numbers.

Reply:
Yeah, I thought about this and anticipated somebody might bring this up. I totally prefer to use durations. In this case, the input duration was a double and already unit-less. Didn't seem worth the extra allocation. And the conversion was covered by a test.